examples/walker/main.go

```diff
@@ -40,12 +51,17 @@ func main() {
 
 type treeVisitor struct{}
 
-func (v *treeVisitor) VisitTree(language core.Language, tree core.ParseTree) error {
+func (v *treeVisitor) VisitTree(languages []core.Language, tree core.ParseTree) error {
 	file, err := tree.File()
 	if err != nil {
 		return fmt.Errorf("failed to get file: %w", err)
 	}
 
+	language, err := tree.Language()
```

Comment:
See the ambiguity here. You are still trusting `tree.Language()`. The `languages` arg is not used which indicates a `TreeVisitor` probably does not need language at all because the same is available from `tree.Language()`

Reply:
my bad, overlooked it
`SourceVisitor` & `TreeVisitor` don't need language to visit tree/file as we already have `tree.Language())`